test/e2e/external_gateways.go

```diff
@@ -924,15 +979,54 @@ var _ = ginkgo.Describe("External Gateway", func() {
 				totalPodConnEntries := pokeConntrackEntries(nodeName, addresses.srcPodIP, protocol, nil)
 				gomega.Expect(totalPodConnEntries).To(gomega.Equal(6)) // total conntrack entries for this pod/protocol
 
-				if deletePod {
+				switch removalType {
+				case GatewayDelete:
 					ginkgo.By(fmt.Sprintf("Delete second external gateway pod %s from ns %s", gatewayPodName2, servingNamespace))
 					err = f.ClientSet.CoreV1().Pods(servingNamespace).Delete(context.TODO(), gatewayPodName2, metav1.DeleteOptions{})
 					framework.ExpectNoError(err, "Delete the gateway pod failed: %v", err)
 					// give some time to handle pod delete event
 					time.Sleep(5 * time.Second)
-				} else {
+				case GatewayUpdate:
 					ginkgo.By("Remove second external gateway pod's routing-namespace annotation")
 					annotatePodForGateway(gatewayPodName2, servingNamespace, "", addresses.gatewayIPs[1], false)
+				case GatewayDeletionTimestamp:
+					ginkgo.By("Setting finalizer then deleting external gateway pod with grace period to set deletion timestamp")
+					p := getGatewayPod(f, servingNamespace, gatewayPodName2)
+					p.Finalizers = append(p.Finalizers, "k8s.ovn.org/external-gw-pod-finalizer")
+					updatePod(f, p)
+					// give some time for the pod update event
+					time.Sleep(5 * time.Second)
```

Comment:
you are not actually validating if the pod is updated with the finalizer, 5s sleep will not guarantee that.